### PR TITLE
Add ray backend for pytorch_lightning 1.6

### PR DESCRIPTION
**Pull request by MeouSker77**
## What changes were proposed in this pull request?

Add `RayStrategy` for pytorch_lightning 1.6.4

## Why are the changes needed?

The api of pytorch_lihgtning 1.6 has changed a lot, we need to make these changes to keep the code working

## Please briefly describe the changes such as API changes and code design.

Add `RayStrategy` and `RayEnvironment`, which will be used when using pytorch_lightning 1.6 and ray backend

## Design Link

#4813 

## How was this patch tested?

unit tests and manual tests

### CODE REVIEW

**Comment by MeouSker77** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/290/

**Comment by MeouSker77** [comment]
main changes:

- refactor `RayPlugin.execution_loop` to `_RayLauncher.launch`
- refactor `RayPlugin.execute_remote` to `_RayLauncher._wrapping_function`
- refactor `RayPlugin.new_process` to `RayStrategy.setup`
- perfect `RayEnvironment`

**Comment by MeouSker77** [comment]
http://10.112.231.51:18888/job/BigDL-NanoTests-PR-Validation/290/

File: python/nano/src/bigdl/nano/pytorch/strategies/ray/ray_distributed.py [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+            ray.get([w.execute.remote(self.init_hook) for w in self.workers])   # type: ignore
+
+    def _configure_launcher(self):
+        self._launcher = _RayLauncher(self)
+
+    def _create_worker(self):
+        """Creates Ray actor."""
+        from bigdl.nano.common.cpu_schedule import schedule_workers
+
+        cpu_procs = schedule_workers(self.num_workers)
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/strategies/ray/ray_distributed.py:239]: you maybe want to look at `schedule_processors` which takes into account the case where the number of workers is larger than available physical cores.

**Comment by yangw1234** [comment]
LGTM except one minor comment (I think it is ok to address in a follow up PR)

**Pull request merged**